File: LeetCode/198 House Robber.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def rob(self, nums):
        ans = []
        for index in range(0, len(nums)):
            money = nums[index]
            if index > 0:
                money = max(money, ans[index - 1])
            if index > 1:
                money = max(money, ans[index - 2] + nums[index])
            ans.append(money)
        if not ans:
            return 0
        return ans[-1]

File: LeetCode/198 House Robber.cpp
class Solution {
 
 public:
  
  int rob(std::vector<int> &nums) {
    std::vector<int> answer = nums;
    const size_t N = nums.size();
    for (size_t i = 0; i < N; ++i) {
      if (i) {
        answer[i] = std::max(answer[i], answer[i - 1]);
      }
      if (i > 1) {
        answer[i] = max(answer[i], answer[i - 2] + nums[i]);
      }
    }
    return N ? answer.back() : 0;
  }

};
